archeryutils/handicaps/handicap_scheme.py

```diff
@@ -0,0 +1,764 @@
+
+        """
+        # Check we have a valid score
+        max_score = rnd.max_score()
+        if score > max_score:
+            raise ValueError(
+                f"The score of {score} provided is greater than the maximum of {max_score} "
+                f"for a {rnd.name}."
+            )
+        if score <= 0.0:
+            raise ValueError(
+                f"The score of {score} provided is less than or equal to zero so cannot "
+                "have a handicap."
+            )
+
+        if score == max_score:
+            # Deal with max score before root finding
+            return self._get_max_score_handicap(rnd, arw_d, int_prec)
+
+        handicap = self._rootfind_score_handicap(score, rnd, arw_d=arw_d)
+
+        # Force integer precision if required.
+        if int_prec:
+            if self.desc_scale:
+                handicap = np.ceil(handicap)
+            else:
+                handicap = np.floor(handicap)
+
+            sc_int = self.score_for_round(rnd, handicap, arw_d, rounded_score=True)
+
+            # Check that you can't get the same score from a larger handicap when
+            # working in integers
+            min_h_flag = False
+            if self.desc_scale:
+                hstep = 1.0
+            else:
+                hstep = -1.0
+            while not min_h_flag:
+                handicap += hstep
+                sc_int = self.score_for_round(rnd, handicap, arw_d, rounded_score=True)
+                if sc_int < score:
+                    handicap -= hstep  # undo the iteration that caused flag to raise
+                    min_h_flag = True
+
+        return handicap
+
+    def _get_max_score_handicap(
+        self,
+        rnd: rounds.Round,
+        arw_d: Optional[float] = None,
+        int_prec: bool = False,
+    ) -> Union[int, float]:
+        """Get handicap for maximum score on a round.
+
+        Start high and drop down until no longer rounding to max score.
+        i.e. >= max_score - 1.0 for ceil() rounding, and >= max_score - 0.5 for around().
+
+        Parameters
+        ----------
+        rnd : rounds.Round
+            round being used
+        arw_d : float or None, default=None
+            user-specified arrow diameter in [metres]
+        int_prec : bool, default=False
+            display results as integers?
+
+        Returns
+        -------
+        handicap : int or float
+            Handicap for maximum score. Has type int if int_prec is True, else float.
+
+        Warns
+        ------
+        UserWarning
+            If called with int_prec=False as value limited by the numerical scheme delta.
+
+        """
+        max_score = rnd.max_score()
+
+        if self.desc_scale:
+            handicap = -75.0
+            delta_hc = 0.01
+        else:
+            handicap = 175.0
+            delta_hc = -0.01
+
+        # Set rounding limit
+        if self.name in ("AGB"):
+            round_lim = 1.0
+        else:
+            round_lim = 0.5
+
```

Comment:
It feels odd that the method on the base class is doing conditional logic looking for a specifc handicap systems name here. 
I'd suggest refactor so that the rounding limit is passed as a parameter to this function or stored as another system variable to be looked up?

Reply:
Agreed, I'd made a mental note to look at this one, but it wasn't as straightforward as the other cases where I swapped out the checks using names for the `desc_scale` attribute.

Reply:
@TomHall2020 Is this now resolved to your satisfaction?